src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.edit-email-address.unit.spec.jsx

```diff
@@ -0,0 +1,244 @@
+    .exist;
+}
+
+// When the initial transaction creation request fails
+async function testTransactionCreationFails() {
+  server.use(
+    ...mocks.updateEmailAddressCreateTransactionFailure,
+    ...mocks.addEmailAddressCreateTransactionFailure,
+  );
+
+  editEmailAddress();
+
+  // expect an error to be shown
+  const alert = await view.findByRole('alert');
+  expect(alert).to.have.class('usa-alert-error');
+  expect(alert).to.contain.text(
+    'We’re sorry. We couldn’t update your email address. Please try again.',
+  );
+}
+
+// When the update fails while the Edit View is still active
+async function testQuickFailure() {
+  server.use(...mocks.editEmailAddressTransactionFailure);
+
+  editEmailAddress();
+
+  // expect an error to be shown
+  const alert = await view.findByRole('alert');
+  expect(alert).to.have.class('usa-alert-error');
+  expect(alert).to.contain.text(
+    'We’re sorry. We couldn’t update your email address. Please try again.',
+  );
+}
+
+// When the update fails but not until after the Edit View has exited and the
+// user returned to the read-only view
+async function testSlowFailure() {
+  server.use(...mocks.editEmailAddressTransactionPending);
+
+  const { emailAddressInput } = editEmailAddress();
+
+  // wait for the edit mode to exit and the new address to show up
+  await waitForElementToBeRemoved(emailAddressInput);
+
+  // check that the "we're saving your..." message appears
+  const savingMessage = await view.findByText(
+    /We’re working on saving your new email address. We’ll show it here once it’s saved./i,
+  );
+  expect(savingMessage).to.exist;
+
+  server.use(...mocks.editEmailAddressTransactionFailure);
+
+  await waitForElementToBeRemoved(savingMessage);
+
+  // make sure the error message appears
+  expect(
+    view.getByText(
+      /We couldn’t save your recent email address update. Please try again later/i,
+    ),
+  ).to.exist;
+
+  // and the new email address should not exist in the DOM
+  expect(view.queryByText(newEmailAddress)).not.to.exist;
+  // and the add email button should be back
+  expect(getEditButton()).to.exist;
+}
+
+describe('Editing email address', () => {
+  before(() => {
+    // before we can use msw, we need to make sure that global.fetch has been
+    // restored and is no longer a sinon stub.
+    resetFetch();
+    server = setupServer(...mocks.editEmailAddressSuccess());
+    server.listen();
+  });
+  beforeEach(() => {
+    window.VetsGov = { pollTimeout: 1 };
+  });
+  afterEach(() => {
+    server.resetHandlers();
+  });
+  after(() => {
+    server.close();
+  });
+
+  describe('when an address does not exist yet', () => {
+    beforeEach(() => {
+      const initialState = createBasicInitialState();
+      initialState.user.profile.vet360.email = null;
+
+      view = renderWithProfileReducers(ui, {
+        initialState,
+      });
+    });
+
+    it('should handle a transaction that succeeds quickly', async () => {
+      await testQuickSuccess();
```

Comment:
@sanlouise when I was sharing my screen, I had code similar to this, _but_ I didn't put `await` in front of the function call. So the `it()` ran but didn't actually wait for all the assertions in `testQuickSuccess()` to run.

Reply:
Thanks to @kelihansen for asking "but were those tests actually passing?"